src/main/kotlin/com/terraformation/backend/accelerator/api/ParticipantProjectSpeciesController.kt

```diff
@@ -30,18 +66,34 @@ class ParticipantProjectSpeciesController(
   ): GetParticipantProjectSpeciesResponsePayload {
     val model = participantProjectSpeciesStore.fetchOneById(participantProjectSpeciesId)
 
-    return GetParticipantProjectSpeciesResponsePayload(
-        ParticipantProjectSpeciesPayload(
-            feedback = model.feedback,
-            id = model.id,
-            projectId = model.projectId,
-            rationale = model.rationale,
-            speciesId = model.speciesId,
-            submissionStatus = model.submissionStatus,
-        ))
+    return makeGetResponse(model)
   }
+
+  private fun makeGetResponse(
+      model: ExistingParticipantProjectSpeciesModel
+  ): GetParticipantProjectSpeciesResponsePayload =
+      GetParticipantProjectSpeciesResponsePayload(
+          ParticipantProjectSpeciesPayload(
+              feedback = model.feedback,
+              id = model.id,
+              projectId = model.projectId,
+              rationale = model.rationale,
+              speciesId = model.speciesId,
+              submissionStatus = model.submissionStatus,
+          ))
 }
 
+data class AssignParticipantProjectSpeciesPayload(
+    val projectIds: List<ProjectId>,
+    val speciesIds: List<SpeciesId>
```

Comment:
I thought about that, but didn't realize we could guarantee the values would be unique since it is a JSON payload. If it indicates the values _should_ be unique, is there any sort of validation or coercion of that requirement when the payload comes in?

Reply:
If you declare this as `Set`, duplicates will be removed (but not rejected as errors) automatically when the JSON is parsed and turned into a payload object. So the actual behavior won't change compared to what's in the PR now; it's more about documenting the intent.